openassessment/xblock/student_training_mixin.py

```diff
@@ -58,6 +58,47 @@ def render_student_training(self, data, suffix=''):   # pylint:disable=W0613
         else:
             return self.render_assessment(path, context)
 
+    def _parse_example(self, example):
+        """
+        EDUCATOR-1263: examples are serialized in a myriad of different ways, we need to be robust to all of them.
+
+        Types of serialized example['answer'] we handle here:
+            -fully specified: {'answer': {'parts': [{'text': <response_string>}]}}
+            -list of string: {'answer': [<response_string>]}
+            -just a string: {'answer': <response_string>}
+        """
+        if example:
+            if isinstance(example, dict):
+                answer = example['answer']
+                if isinstance(answer, dict) and 'parts' in answer and isinstance(answer['parts'], list) \
+                    and len(answer['parts']) > 0 and isinstance(answer['parts'][0], dict) \
+                    and 'text' in answer['parts'][0] and isinstance(answer['parts'][0]['text'], basestring):
+                    return (
+                        create_submission_dict({'answer': answer}, self.prompts),
+                        ""
+                    )
+                elif isinstance(answer, list) and len(answer) > 0 and isinstance(answer[0], basestring):
```

Comment:
The `len(answer) > 0` bit handles the case of empty arrays that bit me this morning.